ikarus/finiteelements/mechanics/materials/neohooke.hh

```diff
@@ -72,11 +72,16 @@ struct NeoHookeT : public Material<NeoHookeT<ST>>
    * \return ScalarType The stored energy.
    */
   template <typename Derived>
-  ScalarType storedEnergyImpl(const Eigen::MatrixBase<Derived>& C) const noexcept {
+  ScalarType storedEnergyImpl(const Eigen::MatrixBase<Derived>& C) const {
     static_assert(Concepts::EigenMatrixOrVoigtNotation3<Derived>);
     if constexpr (!Concepts::EigenVector<Derived>) {
-      const auto traceC  = C.trace();
-      const auto logdetF = log(sqrt(C.determinant()));
+      const auto traceC = C.trace();
+      const auto detC   = C.determinant();
+      if (Dune::FloatCmp::lt(static_cast<double>(detC), 0.0, 1e-10))
+        DUNE_THROW(Dune::MathError,
```

Comment:
I think I got confused again. This should be `InvalidStateException` instead of `MathError` shouldn't it?

Reply:
Yes